app/main.py

```diff
@@ -1,3 +1,73 @@
+from __future__ import annotations
+from typing import Optional
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance" | int | float) -> Optional["Distance"]:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+
+    def __iadd__(
+            self,
+            other: "Distance" | int | float
+    ) -> Optional["Distance"]:
+        if isinstance(other, Distance):
+            self.km += other.km
+
+        if isinstance(other, (int, float)):
+            self.km += other
+
+        return self
+
+    def __mul__(self, other: int) -> Optional["Distance"]:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: int | float) -> Optional["Distance"]:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: "Distance" | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+
+        if isinstance(other, (int, float)):
+            return self.km < other
+
+    def __gt__(self, other: "Distance" | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km > other.km
+
+        if isinstance(other, (int, float)):
+            return self.km > other
+
+    def __eq__(self, other: "Distance" | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+
+        if isinstance(other, (int, float)):
+            return self.km == other
+
+    def __le__(self, other: "Distance" | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km <= other.km
+
+        if isinstance(other, (int, float)):
+            return self.km <= other
+
+    def __ge__(self, other: "Distance" | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km >= other.km
+
+        if isinstance(other, (int, float)):
+            return self.km >= other
```

Comment:
try to rewrite these two methods in one line each. Use previously created magic methods
Hint: <= is the same as not >

Reply:
Nice idea! Thank you for the good advice )